R/table-ident.R

```diff
@@ -0,0 +1,239 @@
+new_table_ident <- function(...,
+                            table = NA_character_,
+                            schema = NA_character_,
+                            catalog = NA_character_,
+                            quoted = FALSE,
+                            error_call = caller_env()) {
+  check_dots_empty(call = error_call)
+
+  check_character(table, call = error_call)
+  check_character(schema, call = error_call)
+  check_character(catalog, call = error_call)
+  check_logical(quoted, call = error_call)
+
+  n <- vctrs::vec_size_common(
```

Comment:
It is not really necessary to support vectors but it might be handy at some point.